src/components/AuthForm.js

```diff
@@ -0,0 +1,142 @@
+import {
+  Form,
+  Link,
+  useSearchParams,
+  useActionData,
+  useNavigation,
+  useNavigate,
+} from "react-router-dom";
+
+import classes from "./AuthForm.module.css";
+import { useState } from "react";
+
+function AuthForm() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const isLogin = searchParams.get("mode") === "login";
+  const [otp, setOtp] = useState(false);
+  const navigation = useNavigation();
+  const isSubmitting = navigation.state === "submitting";
+  const data = useActionData();
+  const navigate = useNavigate();
+
+  async function submitHandler(event) {
+    event.preventDefault();
+    if (!otp) {
+      const data = new FormData(event.target);
+      var object = {};
+      data.forEach((value, key) => (object[key] = value));
+      var body = JSON.stringify(object);
+
+      let url = "http://127.0.0.1:8000/user/";
+
+      if (isLogin) {
+        url += "login/";
+      } else {
+        url += "register_user/";
+      }
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: body,
+      });
+
+      if (!response.ok) {
+        return response;
```

Comment:
Can you please explain this?

Reply:
I think I have made an error here. At first, this code was in actions so returning response like this would send the response to the error page directly and the error page was responsible to show this error. I will change it and create a toast error so that error will be displayed on the same page